File: 2000 98 & ME/QuickStart/winforms/samples/VB/AxHosting/Document.vb
''------------------------------------------------------------------------------
''/ <copyright from='1997' to='2001' company='Microsoft Corporation'>           
''/    Copyright (c) Microsoft Corporation. All Rights Reserved.                
''/       
''/    This source code is intended only as a supplement to Microsoft
''/    Development Tools and/or on-line documentation.  See these other
''/    materials for detailed information regarding Microsoft code samples.
''/
''/ </copyright>                                                                
''------------------------------------------------------------------------------
Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports System.WinForms
Imports AxSHDocVw

Namespace Microsoft.Samples.WinForms.VB.AxHosting 

    Public Class Document
        Inherits System.WinForms.Form

        'Required by the Windows Forms Designer   
        Private components As System.ComponentModel.Container
        Private label1 As System.WinForms.Label
        Private panel1 As System.WinForms.Panel
        Private textBox1 As System.WinForms.TextBox
        Private AxWebBrowser1 As AxSHDocVw.AxWebBrowser
        Private buttonGo As System.WinForms.Button
    
        Public Sub New()
           MyBase.New
    
           'This call is required by the Windows Forms Designer.
           InitializeComponent
    
           ' TODO: Add any constructor code after InitializeComponent call
           Me.Text = "New Web Browser"

           'Handle the Title Change event so that we can update the caption and the 
           'status bar on our parent
           
           AddHandler AxWebBrowser1.TitleChange, AddressOf Me.AxWebBrowser1_TitleChanged

           'Once the Browser Control has been created we can interact with it
           'So hook the handle created event and use  this to navigate 
           'to the start page
           AddHandler AxWebBrowser1.HandleCreated, AddressOf Me.AxWebBrowser1_Created

           'Set the minimum form size 
           Me.MinTrackSize = new Size(100, 100)
        End Sub

        'The WebBrowser has been created so complete initialisation
    	Private Sub AxWebBrowser1_Created(sender As object, evArgs As EventArgs) 
	        
           'Navigate to the starting page
	       buttonGo_Click(buttonGo, EventArgs.Empty)

           'Don't need this event handler any more so remove it 
           RemoveHandler AxWebBrowser1.HandleCreated, AddressOf Me.AxWebBrowser1_Created
            
	    End Sub


        'Handle the title changed event
        Private Sub AxWebBrowser1_TitleChanged(sender As object, e As DWebBrowserEvents2_TitleChangeEvent) 
            Me.Text =  e.p_text
            Dim f As MainForm = CType(Me.MDIParent,MainForm)
            f.statusBar1.Text=Me.Text
        End Sub


        'Handle the click of the go button
        Private Sub buttonGo_Click(sender As object, evArgs As EventArgs) 

            Dim currentCursor As Cursor = Cursor.Current
            Try 
                Cursor.Current = Cursors.WaitCursor

                AxWebBrowser1.Navigate(textBox1.Text,0,"", "", "")

            Finally 
                Cursor.Current = currentCursor 
            End Try
            
        End Sub


        'Clean up any resources being used
        Overrides Public Sub Dispose()
            MyBase.Dispose
            components.Dispose
        End Sub 


        'NOTE: The following procedure is required by the Windows Forms Designer
        'Do not modify it.
        Private Sub InitializeComponent() 
            Me.components = New System.ComponentModel.Container
            Me.textBox1 = New System.WinForms.TextBox
            Me.panel1 = New System.WinForms.Panel
            Me.label1 = New System.WinForms.Label
            Me.AxWebBrowser1 = New AxSHDocVw.AxWebBrowser
            Me.buttonGo = New System.WinForms.Button

            AxWebBrowser1.BeginInit()

            textBox1.Location = New System.Drawing.Point(56, 1)
            textBox1.Text = "http://localhost/quickstart/winforms/samples/cs/axhosting/WinForms.html"
            textBox1.Anchor = System.WinForms.AnchorStyles.LeftRight
            textBox1.BorderStyle = System.WinForms.BorderStyle.FixedSingle
            textBox1.TabIndex = 1
            textBox1.Size = New System.Drawing.Size(504, 20)

            panel1.Dock = System.WinForms.DockStyle.Top
            panel1.Size = New System.Drawing.Size(650, 24)
            panel1.TabIndex = 0
            panel1.Text = "panel1"

            label1.Location = New System.Drawing.Point(0, 4)
            label1.Text = "Address:"
            label1.Anchor = System.WinForms.AnchorStyles.Left
            label1.Size = New System.Drawing.Size(64, 16)
            label1.TabIndex = 0

            Me.AutoScaleBaseSize = New System.Drawing.Size(5, 13)
            '@design Me.TrayLargeIcon = True
            Me.AcceptButton = buttonGo
            '@design Me.TrayHeight = 0
            Me.ClientSize = New System.Drawing.Size(650, 480)

            AxWebBrowser1.Location = New System.Drawing.Point(0, 24)
            AxWebBrowser1.Size = New System.Drawing.Size(650, 456)
            AxWebBrowser1.Dock = System.WinForms.DockStyle.Fill
            AxWebBrowser1.TabIndex = 3

            buttonGo.Location = New System.Drawing.Point(562, 0)
            buttonGo.FlatStyle = System.WinForms.FlatStyle.Flat
            buttonGo.Size = New System.Drawing.Size(88, 24)
            buttonGo.TabIndex = 2
            buttonGo.Anchor = System.WinForms.AnchorStyles.Right
            buttonGo.Text = "&Go"
            AddHandler buttonGo.Click, AddressOf buttonGo_Click

            Me.Controls.Add(panel1)
            panel1.Controls.Add(textBox1)
            panel1.Controls.Add(buttonGo)
            panel1.Controls.Add(label1)
            Me.Controls.Add(AxWebBrowser1)

            AxWebBrowser1.EndInit()
        End Sub

    End Class

End Namespace

File: 2000 98 & ME/QuickStart/winforms/samples/Cs/AxHosting/document.cs
//------------------------------------------------------------------------------
/// <copyright from='1997' to='2001' company='Microsoft Corporation'>           
///    Copyright (c) Microsoft Corporation. All Rights Reserved.                
///       
///    This source code is intended only as a supplement to Microsoft
///    Development Tools and/or on-line documentation.  See these other
///    materials for detailed information regarding Microsoft code samples.
///
/// </copyright>                                                                
//------------------------------------------------------------------------------
namespace Microsoft.Samples.WinForms.Cs.AxHosting {
    using System;
    using System.ComponentModel;
    using System.Drawing;
    using System.WinForms;
    using AxSHDocVw;

    public class Document : System.WinForms.Form {
        private System.ComponentModel.Container components;
        private AxSHDocVw.AxWebBrowser AxWebBrowser1;
        private System.WinForms.Button buttonGo;
        private System.WinForms.TextBox textBox1;
        private System.WinForms.Label label1;
        private System.WinForms.Panel panel1;

        public Document() : base() {
            
            // Required by the Windows Forms Designer
            InitializeComponent();

            // TODO: Add any constructor code after InitializeComponent call
            this.Text = "New Web Browser";

            //Handle the Title Change event so that we can update the caption and the 
            //status bar on our parent
            AxWebBrowser1.TitleChange += new DWebBrowserEvents2_TitleChangeEventHandler(this.AxWebBrowser1_TitleChanged);

            //Once the Browser Control has been created we can interact with it
            //So hook the handle created event and use  this to navigate 
            //to the start page
            AxWebBrowser1.HandleCreated += new EventHandler(this.AxWebBrowser1_Created);
            
            //Set the minimum form size 
            this.MinTrackSize = new Size(100, 100);
        }


        //The WebBrowser has been created so complete initialisation
    	public void AxWebBrowser1_Created(object sender, EventArgs evArgs) {
	        
            //Navigate to the starting page
	        buttonGo_Click(buttonGo, EventArgs.Empty);

            //Don't need this event handler any more so remove it 
            AxWebBrowser1.HandleCreated -= new EventHandler(this.AxWebBrowser1_Created);
	    }


        //Handle the title changed event
        private void AxWebBrowser1_TitleChanged(object sender, DWebBrowserEvents2_TitleChangeEvent e) {
            this.Text =  e.p_text;
            ((MainForm)(this.MDIParent)).statusBar1.Text=this.Text;
        }


        //Handle the click of the go button
        private void buttonGo_Click(object sender, EventArgs evArgs) {

            Cursor currentCursor = Cursor.Current;
            try {
                Cursor.Current = Cursors.WaitCursor;

                // Note Well: The declaration of Navigate in the shdocvw typelib is:
                //
                //   [id(0x00000068), helpstring("Navigates to a URL or file.")]
                //   HRESULT Navigate(
                //                    [in] BSTR URL, 
                //                   [in, optional] VARIANT* Flags, 
                //                   [in, optional] VARIANT* TargetFrameName, 
                //                   [in, optional] VARIANT* PostData, 
                //                   [in, optional] VARIANT* Headers);
                //
                // So today the the tlb importer assumes that the VARIANT* means that this is 
                // an in/out param and so we end up with 
                //   
                //   instance void  Navigate(String uRL,
                //                ref Object flags,
                //                ref Object targetFrameName,
                //                ref Object postData,
                //                ref Object& headers) 
                //
                //  This will be fixed so that the tlb importer examines the paramter
                //  attributes post Beta1, thus giving the more "natural"
                //
                //    instance void  Navigate(String uRL,       
                //                 Object flags,            
                //                 Object targetFrameName,  
                //                 Object postData,         
                //                 Object& headers)         
                // 
                //  This explains why wer are passing reference objects to navigate below
                //
                object arg1 = 0; object arg2 = ""; object arg3 = ""; object arg4 = "";
				if (textBox1.Text == null) {
					Console.WriteLine("textBox1.Text = null");
					textBox1.Text = "";
				}
                AxWebBrowser1.Navigate(textBox1.Text,ref arg1,ref arg2, ref arg3, ref arg4);
                
                //Post PDC: AxWebBrowser1.Navigate(textBox1.Text,0,"", "", "");

            } finally {
                Cursor.Current = currentCursor ;
            }
        }


        public override void Dispose() {
            base.Dispose();
            components.Dispose();
        }


        private void InitializeComponent() {

            this.components = new System.ComponentModel.Container();
            this.AxWebBrowser1 = new AxSHDocVw.AxWebBrowser();
            this.textBox1 = new System.WinForms.TextBox();
            this.panel1 = new System.WinForms.Panel();
            this.label1 = new System.WinForms.Label();
            this.buttonGo = new System.WinForms.Button();

            AxWebBrowser1.BeginInit();
            AxWebBrowser1.Size = new System.Drawing.Size(292, 273);
            AxWebBrowser1.TabIndex = 3;
            AxWebBrowser1.Dock = System.WinForms.DockStyle.Fill;

            textBox1.Text = "http://localhost/quickstart/winforms/samples/cs/axhosting/WinForms.html";
            textBox1.Anchor = System.WinForms.AnchorStyles.LeftRight;
            textBox1.TabIndex = 1;
            textBox1.Size = new System.Drawing.Size(450, 20);
            textBox1.Location = new System.Drawing.Point(56, 2);
            textBox1.BorderStyle = System.WinForms.BorderStyle.FixedSingle;


            panel1.Dock = System.WinForms.DockStyle.Top;
            panel1.AutoScrollMinSize = new System.Drawing.Size(0, 0);
            panel1.Size = new System.Drawing.Size(600, 24);
            panel1.TabIndex = 0;
            panel1.Text = "panel1";

            label1.Size = new System.Drawing.Size(64, 16);
            label1.Anchor = System.WinForms.AnchorStyles.Left;
            label1.TabIndex = 0;
            label1.Text = "Address:";
            label1.Location = new System.Drawing.Point(0, 4);

            buttonGo.Anchor = System.WinForms.AnchorStyles.Right;
            buttonGo.Size = new System.Drawing.Size(88, 24);
            buttonGo.TabIndex = 2;
            buttonGo.Text = "&Go";
            buttonGo.Location = new System.Drawing.Point(510, 0);
            buttonGo.FlatStyle = System.WinForms.FlatStyle.Flat;
            buttonGo.Click += new System.EventHandler(buttonGo_Click);

            this.AutoScaleBaseSize = new System.Drawing.Size(5, 13);
            this.Text = "";
            this.ClientSize = new System.Drawing.Size(650, 480);
            this.AcceptButton = buttonGo;

            this.Controls.Add(AxWebBrowser1);
            this.Controls.Add(panel1);
            panel1.Controls.Add(textBox1);
            panel1.Controls.Add(buttonGo);
            panel1.Controls.Add(label1);
            this.Controls.Add(AxWebBrowser1);

            AxWebBrowser1.EndInit();

	    }


    }
}










